File: 0796-rotate-string/0796-rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
        if(s.equals(goal)){
            return true;
        }
        int n=s.length();
        int k=1;
        char[] temp=s.toCharArray();
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                temp[(j+k)%n]=s.charAt(j);
            }
            String temp1=new String(temp);
            if(temp1.equals(goal)){
                return true;
            }
            k++;
        }
        return false;
    }
}

File: 0796-rotate-string/0796-rotate-string.c
bool rotateString(char* s, char* goal) {
    int r=1;
    int n=strlen(s);
    if(n!=strlen(goal)){
        return false;
    }
    if(strcmp(goal,s)==0){
        return true;
    }
    while(r<n){
        char temp[n+1];
        for(int i=0;i<n;i++){
            temp[(i+r)%n]=s[i];
        }
        temp[n]='\0'; //Imp note it
        if(strcmp(temp,goal)==0){
            return true;
        }
        r++;
    }
    return false;
}